app/jobs/operation_job.rb

```diff
@@ -3,53 +3,58 @@ class OperationJob < ApplicationJob
 
   rescue_from StandardError, with: :operation_raised_error
 
-  before_perform { |job| job.operation_started }
-
-  after_perform { |job| job.operation_completed }
+  before_perform :operation_started, if: :operation
+  before_perform :load_settings, if: :operation
+  after_perform :operation_completed, if: :operation
 
   protected
 
-    def operation
-      arguments.first.tap do |operation|
-        if operation.nil?
-          Rails.logger.warn "Unable to update failed operation for job: #{self.inspect}"
-          break
-        end
-
-        unless operation.is_a?(Operation)
-          raise ArgumentError, "unexpected type for operation: #{operation.inspect}"
-        end
-      end
-    end
-
-    def operation_started
-      operation.present? && operation.update_attribute(:job_started_at, Time.now)
-    end
-
-    def operation_succeeded(outcome_url=nil)
-      operation.present? && operation.update_attribute(:job_outcome_url, outcome_url) if outcome_url.present?
-    end
-
-    def operation_failed(report)
-      return unless operation.present?
-      save_failure(report)
-    end
-
-    def operation_raised_error(exception)
-      return unless operation.present?
-      ExceptionNotifier.notify_exception(exception)
-      save_failure(I18n.t("operation.server_error"))
-    end
-
-    def operation_completed
-      operation.present? && operation.update_attribute(:job_completed_at, Time.now)
-    end
+  def operation
+    # The `Operation` instance tracking this job is always passed as
+    # the first argument to `perform`
+    arguments.first
+  end
+
+  def mission
+    # The `Mission` instance is currently passed as the second arg
+    # to all `OperationJob`s.
+    # TODO: once an operation is associated with a mission this should
+    # be changed to `operation.mission` and operation job argument lists
+    # should be updated to omit the mission
```

Comment:
This should also become reality before we merge this one. So let's plan to change this over before merge.

Reply:
Yup, I'm about to merge the other PR.